src/game/AI/ScriptDevAI/scripts/outland/hellfire_citadel/shattered_halls/shattered_hallsScripts.cpp

```diff
@@ -270,6 +269,267 @@ struct npc_shattered_hand_centurion : public CombatAI
+                }
+            }
+            m_reinfCD = true;
+        }
+
+        // Legionnaire 006
+        if (guid == SIX_LEGIONNAIRE_GUID)
+        {
+            // there are 4 sleeping npcs around him, if one of his group members dies he will call for one of the sleeping creatures to get up and join the fight
+            // this doesnt have a cd, if all 4 npcs with sleeping aura are up, nothing more happens
+            auto worldObjects = m_creature->GetMap()->GetWorldObjects(SLEEPING_REINF_STRING);
+            WorldObject* closest = nullptr;
+            for (WorldObject* wo : *worldObjects)
+            {
+                if (wo->IsCreature() && static_cast<Creature*>(wo)->IsAlive() && static_cast<Creature*>(wo)->HasAura(AURA_SLEEPING))
+                    continue;
+
+                if (!closest)
+                    closest = wo;
+                else if (m_creature->GetDistance(wo, true, DIST_CALC_NONE) < m_creature->GetDistance(closest, true, DIST_CALC_NONE))
+                    closest = wo;
+            }
+            if (closest)
+            {
+                DoBroadcastText(aRandomReinfSleeping[urand(0, 6)], m_creature);
+                static_cast<Creature*>(closest)->RemoveAurasDueToSpell(AURA_SLEEPING);
+                static_cast<Creature*>(closest)->GetMotionMaster()->MovePoint(1, FelOrcSpawnCoords[legionnaireGuid][0], FelOrcSpawnCoords[legionnaireGuid][1], FelOrcSpawnCoords[legionnaireGuid][2], FORCED_MOVEMENT_RUN);
+            }
+        }
+        else if (guid == SEVENTH_LEGIONNAIRE_GUID)
+        {
+            // For the legionnaire 07, if one of his group members dies he will inform the nearest npc staying at the dummys behind him
+            // all 3 npc at dummys have StringID 5400015 assigned
+            auto worldObjects = m_creature->GetMap()->GetWorldObjects(DUMMY_REINF_STRING_1);
+            WorldObject* closest = nullptr;
+            for (WorldObject* wo : *worldObjects)
+            {
+                if (wo->IsCreature() && static_cast<Creature*>(wo)->IsAlive())
+                    continue;
+
+                if (!closest)
+                    closest = wo;
+                else if (m_creature->GetDistance(wo, true, DIST_CALC_NONE) < m_creature->GetDistance(closest, true, DIST_CALC_NONE))
+                    closest = wo;
+            }
+            if (closest)
+                if (m_creature->IsInCombat())
+                {   
+                    static_cast<Creature*>(closest)->AI()->AttackClosestEnemy();
+                    DoBroadcastText(aRandomReinf[urand(0, 6)], m_creature);
+                }
+        }
+        else if (guid == EIGTH_LEGIONNAIRE_GUID)
+        {
+            // For the legionnaire 08, if one of his group members dies he will inform the nearest npc staying at the dummys behind him
+            // all 3 npc at dummys have StringID 5400016 assigned
+            auto worldObjects = m_creature->GetMap()->GetWorldObjects(DUMMY_REINF_STRING_2);
```

Comment:
Func exists for GetCreatures,  so you do not have to deal with the pesky casts and stuff. It will still contain  the same thing ofc.

Reply:
GetWorldObjects exists mainly for scenarios like dbscripts where I do not know what I will be handed.